b1ddi/data_source_dns_forward_zone.go

```diff
@@ -0,0 +1,98 @@
+package b1ddi
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/go-openapi/swag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	b1ddiclient "github.com/infobloxopen/b1ddi-go-client/client"
+	"github.com/infobloxopen/b1ddi-go-client/dns_config/forward_zone"
+	"github.com/infobloxopen/b1ddi-go-client/models"
+)
+
+func dataSourceConfigForwardZone() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceConfigForwardZoneRead,
+		Schema: map[string]*schema.Schema{
+			"filters": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Configure a map of filters to be applied on the search result.",
+			},
+			"results": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        dataSourceSchemaFromResource(resourceConfigForwardZone),
+				Description: "List of DNS Forward Zones matching filters. The schema of each element is identical to the b1ddi_dns_forward_zone resource schema.",
+			},
+		},
+	}
+}
+
+func dataSourceConfigForwardZoneRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*b1ddiclient.Client)
+
+	var diags diag.Diagnostics
+
+	filtersMap := d.Get("filters").(map[string]interface{})
+	filterStr := filterFromMap(filtersMap)
+
+	resp, err := c.DNSConfigurationAPI.ForwardZone.ForwardZoneList(&forward_zone.ForwardZoneListParams{
+		Filter:  swag.String(filterStr),
+		Context: ctx,
+	}, nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	results := make([]interface{}, 0, len(resp.Payload.Results))
+	for _, ab := range resp.Payload.Results {
+		results = append(results, flattenConfigForwardZone(ab)...)
+	}
+	err = d.Set("results", results)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+
+	// always run
+	d.SetId(strconv.FormatInt(time.Now().Unix(), 10))
+
+	return diags
+}
+
+func flattenConfigForwardZone(r *models.ConfigForwardZone) []interface{} {
+	if r == nil {
+		return nil
+	}
+
+	externalForwarders := make([]map[string]interface{}, 0, len(r.ExternalForwarders))
+	for _, ep := range r.ExternalForwarders {
```

Comment:
I guess `ep` comes from a copy/paste from auth zone and external primaries. Using `ef` would be better.

Reply:
Updated